### PR TITLE
Python wrapper improvement

### PR DESCRIPTION
**Pull request by AlexanderSaydakov**
added iterators, rearranged and simplified existing code

### CODE REVIEW

File: python/tests/vo_test.py [hunk 1]
```diff
@@ -45,6 +45,13 @@ def test_vo_example(self):
     items = vo.get_samples()
     self.assertEqual(len(items), k)
 
+    count = 0
+    for tuple in vo:
+      sample = tuple[0]
+      weight = tuple[1]
```
Comment (jmalkin) [inline python/tests/vo_test.py:51]: worth summing weights and ensuring they equal the total?

**Review by jmalkin — COMMENTED** [review]
There are still some C-style function pointer casts -- dunno if we wanna find them all first? And a few of the lambdas are complex enough that I'd have kept them as separate methods.

Also, I think we should wait until we fix the accidental push to the main branch before merging as it'll be a little cleaner to untangle that way.

File: python/src/fi_wrapper.cpp [hunk 2]
```diff
@@ -105,15 +53,55 @@ void bind_fi_sketch(py::module &m, const char* name) {
...
-    .def_static("get_epsilon_for_lg_size", &dspy::fi_sketch_get_generic_epsilon<T>, py::arg("lg_max_map_size"),
-         "Returns the epsilon value used to compute a priori error for a given log2(max_map_size)")
-    .def_static("get_apriori_error", &frequent_items_sketch<T>::get_apriori_error, py::arg("lg_max_map_size"), py::arg("estimated_total_weight"),
-         "Returns the estimated a priori error given the max_map_size for the sketch and the estimated_total_stream_weight.")
-    .def("get_serialized_size_bytes", &dspy::fi_sketch_get_serialized_size_bytes<T>,
-         "Computes the size needed to serialize the current state of the sketch. This can be expensive since every item needs to be looked at.")
-    .def("serialize", &dspy::fi_sketch_serialize<T>, "Serializes the sketch into a bytes object")
-    .def_static("deserialize", &dspy::fi_sketch_deserialize<T>, "Reads a bytes object and returns the corresponding frequent_strings_sketch")
-    ;
+    .def(
```
[the remark this answers is not in the document]
Comment (AlexanderSaydakov) [reply python/src/fi_wrapper.cpp:56]: I thought that getting rid of that large section with namespaces would be good. Yes, this particular function is not trivial, but in my view it is not worse than having that large section just for this function.

File: python/tests/vo_test.py (hunk 1 above)
Comment (AlexanderSaydakov) [reply python/tests/vo_test.py:51]: This seems to be a bit tricky since it is not exact. So I would suggest not test sketch correctness here.

Comment (jmalkin) [reply python/tests/vo_test.py:51]: Discussed offline. It's exact modulo machine precision with the arithmetic, but that's a fair point. It is reasonable to test that part only in C++.

**Review by jmalkin — APPROVED** [review]
LGTM

**Pull request merged**